debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -81,6 +81,9 @@ public R apply(R r) {
         // Ignoring messages which do not adhere to the CDC Envelope, for instance:
         // Heartbeat and Schema Change messages
         if (!smtManager.isValidEnvelope(r)) {
+            if (smtManager.isHeartbeat(r)) {
+                return null;
```

Comment:
Why is it that heartbeats are skipped? They should be passed on unmodified (i.e. as it was before), as otherwise people would e.g. run into WAL growth issues.

Reply:
Perhaps I misunderstood the issue?  The reporter was using {{ByteBufferConverter}} and this was throwing an error when Kafka was applying the converter to the Heartbeat messages after the transformation by the Outbox.  By returning null from the SMT, this would have prevented the converter from being invoked and avoided the failure.  If that's not the right approach, what should I have done instead?

Reply:
You're right, there's more to it than I first thought.

We cannot simply filter out any messages, though. While it might be ok-ish for heartbeats, it'd be unexpected e.g. for TX metadata events or schema change events. Instead, we'll likely have to follow a similar approach as we do for the CloudEvents converter: let it delegate to another converter (JSON, Avro). Specifically, if the `ByteBufferConverter` encounters something other than `BYTES`, it should delegate conversion of that record to the configured delegate converter. Then, heartbeats (and any other non-CDC events) can be passed on by the outbox routing SMT unmodified. Would that make sense?

Reply:
It does, thanks for the clarification.